napari/components/viewer_model.py

```diff
@@ -484,13 +484,16 @@ def _on_active_layer(self, event):
         if active_layer is None:
             for layer in self.layers:
                 layer.update_transform_box_visibility(False)
+                layer.update_highlight_visibility(False)
```

Comment:
Maybe could it be worthy to unify these calls over a Layer method, something like a `_on_active_layer(is_active_layer: bool)` method?

Reply:
hmm it's not a bad idea, but I I feel like the layer shouldn't actually know whether it's active or not? That seems leaky to me but maybe others disagree.

Reply:
Yeah, I think we are still struggling a bit figuring out where the "boundaries" between data, metadata, layers, and the UI are. And it's a high priority to come up with a consistent model of that.

But we do want *somewhere* to store this. It just might be a higher-level "LayerUI" object?